spec/resolver/basic_spec.rb

```diff
@@ -100,8 +100,82 @@
       deps << Bundler::DepProxy.new(d, "ruby")
     end
 
-    got = Bundler::Resolver.resolve(deps, @index, {}, [], Bundler::RubyVersion.new("1.8.7", nil, nil, nil))
-    got = got.map(&:full_name).sort
-    expect(got).to eq(%w(foo-1.0.0 bar-1.0.0).sort)
+    should_resolve_and_include %w(foo-1.0.0 bar-1.0.0), [{}, [], Bundler::RubyVersion.new("1.8.7", nil, nil, nil)]
+  end
+
+  context "conservative" do
+    before :each do
+      @index = build_index do
+        gem("foo", "1.3.7") { dep "bar", "~> 2.0" }
+        gem("foo", "1.3.8") { dep "bar", "~> 2.0" }
+        gem("foo", "1.4.3") { dep "bar", "~> 2.0" }
+        gem("foo", "1.4.4") { dep "bar", "~> 2.0" }
+        gem("foo", "1.4.5") { dep "bar", "~> 2.1" }
+        gem("foo", "1.5.0") { dep "bar", "~> 2.1" }
+        gem("foo", "1.5.1") { dep "bar", "~> 3.0" }
+        gem "bar", %w(2.0.3 2.0.4 2.0.5 2.1.0 2.1.1 3.0.0)
+      end
+      dep "foo"
+
+      @locked = locked(%w(foo 1.4.3), %w(bar 2.0.3))
+    end
+
+    it "resolves all gems to latest patch" do
+      # strict is not set, so bar goes up a minor version due to dependency from foo 1.4.5
+      should_consv_resolve_and_include :patch, [], %w(foo-1.4.5 bar-2.1.1)
+    end
+
+    it "resolves all gems to latest patch strict" do
+      # strict is set, so foo can only go up to 1.4.4 to avoid bar going up a minor version, and bar can go up to 2.0.5
+      should_consv_resolve_and_include [:patch, :strict], [], %w(foo-1.4.4 bar-2.0.5)
+    end
+
+    it "resolves all gems to latest patch minimal" do
+      # minimal is set, so foo goes up the next available to 1.4.4 and bar goes up to next available 2.0.4
+      should_consv_resolve_and_include [:patch, :minimal], [], %w(foo-1.4.4 bar-2.0.4)
+    end
+
+    it "resolves foo only to latest patch - same dependency case" do
+      @locked = locked(%w(foo 1.3.7), %w(bar 2.0.3))
+      # bar is locked, and the lock holds here because the dependency on bar doesn't change on the matching foo version.
+      should_consv_resolve_and_include :patch, ["foo"], %w(foo-1.3.8 bar-2.0.3)
+    end
+
+    it "resolves foo only to latest patch - changing dependency case" do
```

Comment:
just thinking through this - here's the potential design flaw. In all other cases, `strict` means: ensure no _unlocking_ gem gets past the `patch`/`minor` threshold. And accordingly, not specifying `strict` means for _unlocking_ gems, if one of them has to go past the `patch`/`minor` threshold to satisfy the dependency tree, then ... ok.

But in THIS case, the 'not specifying `strict`' case means for _unlocking gems *and their dependencies*_ ... and thus having to use `strict` to get the expected behavior is a debatable flaw. 

Compare this case to the prior spec: "resolves foo only to latest patch - same dependency case" - that should highlight the inconsistency.

I'm not sure there's an easy code fix for this case, though.